hackathon_site/dashboard/frontend/src/slices/hardware/cartSlice.ts

```diff
@@ -26,6 +30,73 @@ export const cartReducerName = "cart";
 export const initialState = cartAdapter.getInitialState(extraState);
 export type CartState = typeof initialState;
 
+// Thunk
+interface RejectValue {
+    status: number;
+    message: string[];
+}
+
+export interface OrderResponse {
+    order_id: number;
+    hardware: [
+        {
+            hardware_id: number;
+            quantity_fulfilled: number;
+        }
+    ];
+    errors:
+        | [
+              {
+                  hardware_id: number;
+                  message: string;
+              }
+          ]
+        | [];
+}
+
+export const submitOrder = createAsyncThunk<
+    APIListResponse<OrderResponse>,
```

Comment:
Back to what we were talking about, can you change this to `AxiosResponse<OrderResponse>`